CMakeLists.txt

```diff
@@ -1,10 +1,275 @@
 # Native code side of Thrive
-cmake_minimum_required(VERSION 3.10)
+cmake_minimum_required(VERSION 3.13)
+
+set(CMAKE_CONFIGURATION_TYPES "Debug;Release;RelWithDebInfo;Distribution" CACHE STRING "" FORCE)
+set_property(GLOBAL PROPERTY USE_FOLDERS ON)
+
+# Force x64 for Windows before project() call
+if(WIN32)
+    set(CMAKE_GENERATOR_PLATFORM x64)
+endif()
 
 project(Thrive)
 
 # If you want to get compile commands run cmake with
 # "-DCMAKE_EXPORT_COMPILE_COMMANDS=ON"
+# Options
+# Building either the faster variant with AVX or without for older CPU support
+
+# CPU core optimization 
```

Comment:
All of this stuff is super long, so I think it should be put into a separate file like `Scripts/CMakeTargetConfiguration.cmake` and included here.

Reply:
you mean keeping all the compile as is for Linux and just use .cmakes for windows in the implementation you suggested?

Reply:
No, I mean putting all of the long code you plan to keep into a separate file so that reading the main cmake file doesn't require you to read through a ton of specific details before getting to the options etc. actually important parts of the file.